### DISCUSSION TITLE
Encoding DV MKV with DGSource

### DISCUSSION THREAD

**Discussion by cartman0208** [staxrip/staxrip#1250] [Q&A] [answered]
Hiho,
I'm trying to reencode some Dolby Vision MKV exported from MakeMKV with DGSource as source filter
MediaInfo looks like this:
HDR format                               : Dolby Vision, Version 1.0, Profile 7.6, dvhe.07.06, BL+EL+RPU, Blu-ray compatible / SMPTE ST 2086, Version HDR10, HDR10 compatible

But I can't get Staxrip to demux the HDR Metadata
I also noticed that the variable of %source_file% changes from "source.mkv" to "source.dgi" somewhere in the process.

Am I missing some setting?
DGSource is set to index, no demux.

**Comment by Dendraspis** [comment]
`DGSource` creates a `.d2v` file that is used as source file instead and this file has not the right media info, so StaxRip cannot extract anything from it.

The easiest way is to use a default or another preprocessor. Otherwise extract and insert the metadata manually.

If you want StaxRip to do the work, you could select the default preprocessor and open the source file so all metadata is extracted, then switch back to `DGSource`, load the source file again and because the metadata is found, it should be used.

**Reply by cartman0208** [reply]
Thanks, that might work.
So as long as there is file with the correct extension ( .bin for DV) and/or ( .json for HDR10) in the working folder, StaxRip recognizes it in the VUI section?

What would be the default preprocessor?

**Reply by Dendraspis** [reply, marked as answer]
The default settings should be letting `mkvextract` dealing with the `mkv` file:
![image](https://github.com/staxrip/staxrip/assets/50659978/caa5d4a9-eb49-4001-90e8-16ac0e1ba2cb)

The question would be what's inside the `mkv` file...

> So as long as there is file with the correct extension ( .bin for DV) and/or ( .json for HDR10) in the working folder, StaxRip recognizes it in the VUI section?

In the `_temp` folder, when the file is being opened, yes.
Or next to the source file starting with the same name as the source file.

**Reply by cartman0208** [reply]
Ok, that works ... I created 2 project templates, one for demuxing, the other for ... all the rest :-)
